Tapping a list item opens the details

chelseybaker opened the pull request:
Using a fragment!

stkent commented:
Uh oh, a crash!

Steps to reproduce:

- Launch app
- Enter any valid ZIP code
- Tap 'Find Location (List)' button
- Select any entry
- Rotate device

Stack trace is copied below; should be a 100% reliable reproduction though.

```
06-17 15:58:16.432 26996-26996/com.detroitlabs.chelseybaker.wundergroundweather E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                                  Process: com.detroitlabs.chelseybaker.wundergroundweather, PID: 26996
                                                                                                  java.lang.RuntimeException: Unable to start activity ComponentInfo{com.detroitlabs.chelseybaker.wundergroundweather/com.detroitlabs.chelseybaker.wundergroundweather.activities.WeatherDetailsActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.detroitlabs.chelseybaker.wundergroundweather.models.ForecastDay.getTitle()' on a null object reference
                                                                                                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
                                                                                                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
                                                                                                      at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4077)
                                                                                                      at android.app.ActivityThread.-wrap15(ActivityThread.java)
                                                                                                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350)
                                                                                                      at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                                      at android.os.Looper.loop(Looper.java:148)
                                                                                                      at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                                                      at java.lang.reflect.Method.invoke(Native Method)
                                                                                                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                                                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                                                   Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.detroitlabs.chelseybaker.wundergroundweather.models.ForecastDay.getTitle()' on a null object reference
                                                                                                      at com.detroitlabs.chelseybaker.wundergroundweather.fragements.ForecastDetailsPageFragment.onCreateView(ForecastDetailsPageFragment.java:37)
                                                                                                      at android.support.v4.app.Fragment.performCreateView(Fragment.java:1974)
                                                                                                      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
                                                                                                      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1252)
                                                                                                      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1234)
                                                                                                      at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2046)
                                                                                                      at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:174)
                                                                                                      at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:597)
                                                                                                      at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1237)
                                                                                                      at android.app.Activity.performStart(Activity.java:6268)
                                                                                                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2379)
                                                                                                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) 
                                                                                                      at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4077) 
                                                                                                      at android.app.ActivityThread.-wrap15(ActivityThread.java) 
                                                                                                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350) 
                                                                                                      at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                                      at android.os.Looper.loop(Looper.java:148) 
                                                                                                      at android.app.ActivityThread.main(ActivityThread.java:5417) 
                                                                                                      at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                                                                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 
```

chelseybaker commented:
Fixed the rotation problems. 

I learned that rotation causes the activities and fragments to be recreated which I find odd. I wonder what the benefit is to that because it seems like a rotation should definitely not dump all the data and restart. 

I implemented `onSaveInstanceState` and stored the `forecastDay` property, then pull it out again if it's there in `onCreateView`

stkent commented on app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/fragements/ForecastDetailsPageFragment.java:
```diff
@@ -29,24 +31,32 @@ public static ForecastDetailsPageFragment newInstance(@NonNull ForecastDay forec
+    @Override
+    public void onSaveInstanceState(final Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putParcelable(FORECAST_DAY, this.forecastDay);
```
Let's go ahead and replace this with fragment arguments since that's more of a platform standard. Once that's done, LGTM!
